sphinx/ext/autosummary/generate.py

```diff
@@ -221,12 +217,19 @@ def find_autosummary_in_files(filenames):
     """
     documented = []
     for filename in filenames:
-        f = open(filename, 'r')
-        lines = f.read().splitlines()
-        documented.extend(find_autosummary_in_lines(lines, filename=filename))
-        f.close()
+        with open(filename, 'rb') as f:
+            lines = f.read()
+            try:
+                # sometimes this fails (e.g., Windows Py3k w/utf-8 files)
+                lines = lines.decode()
+            except Exception:
+                lines = lines.decode('utf-8')
```

Comment:
I've never used that one before, I'll swap it in. Just for my information, though, why is it preferable to `.decode('utf-8', errors='ignore')`? Because you don't have to append a `.decode(...)` on every instance of `fid.read()`?